Overview page sdk (#853)
opsmill/infrahub-sdk-python: docs/docs/python-sdk/introduction.mdx

<Tabs groupId="async-sync">
<TabItem value="Async" default>

```python
import asyncio
from infrahub_sdk import InfrahubClient

async def main():
    client = InfrahubClient()

    # Create a new tag and save it
    tag = await client.create(
        kind="BuiltinTag",
        name="staging",
        description="Resources in the staging environment",
    )
    await tag.save()

    # Query all tags
    tags = await client.all(kind="BuiltinTag")
    for tag in tags:
        print(f"{tag.name.value} — {tag.description.value}")

asyncio.run(main())

```

</TabItem>
<TabItem value="Sync">

```python
from infrahub_sdk import InfrahubClientSync

client = InfrahubClientSync()

# Create a new tag and save it
tag = client.create(
    kind="BuiltinTag",
    name="staging",
    description="Resources in the staging environment",
)
tag.save()

# Query all tags
tags = client.all(kind="BuiltinTag")
for tag in tags:
    print(f"{tag.name.value} — {tag.description.value}")

```

</TabItem>
</Tabs>

Both async and sync clients expose the same API. Choose async for applications that benefit from concurrent I/O (transforms, large-scale sync scripts) and sync for straightforward scripting.

> ## Why use the SDK over direct API calls

Author: ```suggestion
## When to use the SDK instead of direct API calls
```
